frontend/src/main/web/lib/stores/GlossaryStore.js

```diff
@@ -0,0 +1,522 @@
+import _ from 'lodash';
+
+var SIZE_PER_PAGE = 5000, CHANGE_EVENT = "change",
+    MAX_LISTENER = 100; //number of listener for GlossaryStore (default is 11)
+
+EventEmitter.prototype.setMaxListeners(MAX_LISTENER);
+
+var _state = {
+  canAddNewEntry: false,
+  canUpdateEntry: false,
+  localeOptions: [],
+  srcLocale: null,
+  selectedTransLocale: null,
+  locales: null,
+  glossary: {},
+  original_glossary: {},
+  glossaryResId: [],
+  page:1,
+  filter: '',
+  focusedRow: {
+    rowIndex: -1
+  },
+  uploadFile: {
+    show: false,
+    status : -1,
+    file: null,
+    transLocale: null
+  },
+  sort: {
+    src_content: true
+  },
+  totalCount: 0,
+  newEntry: {
+    term: '',
+    pos: '',
+    description: '',
+    isSaving: false,
+    show: false,
+    isAllowSave: false
+  }
+};
+
+function localesStatAPIUrl() {
+  return Configs.baseUrl + "/glossary/info" + Configs.urlPostfix
+}
+
+function loadLocalesStats() {
+  var url = localesStatAPIUrl();
+
+  return new Promise(function(resolve, reject) {
+    Request.get(url)
+      .set("Cache-Control", "no-cache, no-store, must-revalidate")
+      .set('Accept', 'application/json')
+      .set("Pragma", "no-cache")
+      .set("Expires", 0)
+      .end(function (err, res) {
+        if(err != null && err.error === true) {
+          console.error(url, err);
+        }
+        if(res != null) {
+          if (res.error) {
+            console.error(url, res.status, res.error.toString());
+            reject(Error(res.error.toString()));
+          } else {
+            resolve(res['body']);
+          }
+        }
+        resolve(null);
+      })
+  });
+}
+
+function processLocalesStatistic(serverResponse) {
+  _state['locales'] = {};
+  _state['localeOptions'] = [];
+  _state['srcLocale'] = serverResponse['srcLocale'];
+  resetCache();
+
+  _.forEach(serverResponse['transLocale'], function(transLocale) {
+    _state['locales'][transLocale.locale.localeId] = transLocale;
+    _state['localeOptions'].push({
+      value: transLocale.locale.localeId,
+      label: `${transLocale.locale.displayName} - (${transLocale.numberOfTerms})`
+    });
+  });
+  return _state;
+}
+
+function glossaryAPIUrl(srcLocaleId, transLocale) {
+  var page = _state['page'], filter = _state['filter'];
+
+  var url = Configs.baseUrl + "/glossary/src/" + srcLocaleId;
+
+  if(!StringUtils.isEmptyOrNull(transLocale)) {
+    url = url + "/trans/" + transLocale;
+  }
+  url = url + Configs.urlPostfix + "?page=" + page + "&sizePerPage=" + SIZE_PER_PAGE;
+
+  if(!StringUtils.isEmptyOrNull(filter)) {
+    url = url + "&filter=" + filter;
+  }
+  return url + generateSortOrderParam();
+}
+
+
+function generateSortOrderParam() {
+  var params = [];
+  _.forOwn(_state['sort'], function (value, field) {
+    var param = value === true ? field : "-" + field;
+    params.push(param);
+  });
+
+  return _.size(params) > 0 ? "&sort=" + params.join() : '';
+}
+
+function loadGlossaryByLocale () {
```

Comment:
Could extract most of this function to a separate module (to make GlossaryStore less huge) - just need a function that takes the URL and returns a promise.